### PR TITLE
test improvements

### PR DESCRIPTION
**Pull request by Maks3w**

### CODE REVIEW

File: test/MessageTraitTest.php [hunk 1]
```diff
@@ -186,13 +183,6 @@ public function testWithoutHeaderDoesNothingIfHeaderDoesNotExist()
         $this->assertFalse($message->hasHeader('X-Foo'));
     }
 
-    public function testHeadersInitialization()
-    {
-        $headers = ['X-Foo' => ['bar']];
-        $this->message = new Request(null, null, $this->stream, $headers);
-        $this->assertSame($headers, $this->message->getHeaders());
-    }
```
Comment (weierophinney) [inline test/MessageTraitTest.php]: Why did you remove this test?

File: test/MessageTraitTest.php [hunk 2]
```diff
@@ -33,14 +35,9 @@ public function testProtocolMutatorReturnsCloneWithChanges()
         $this->assertEquals('1.0', $message->getProtocolVersion());
     }
 
-    public function testUsesStreamProvidedInConstructorAsBody()
-    {
-        $this->assertSame($this->stream, $this->message->getBody());
-    }
```
Comment (weierophinney) [inline test/MessageTraitTest.php]: Why did you remove this test?

File: test/ServerTest.php [hunk 3]
```diff
@@ -129,7 +136,7 @@ public function testListenEmitsStatusHeaderWithoutReasonPhraseIfNoReasonPhrase()
             'QUERY_STRING' => 'bar=baz',
         ];
 
-        $callback = function ($req, $res) {
+        $callback = function (ServerRequestInterface $req, ResponseInterface $res) {
```
Comment (weierophinney) [inline test/ServerTest.php]: The typehints are strictly necessary here; why are you adding them?

Comment (weierophinney) [inline test/ServerTest.php]: I'm assuming it's for IDE support, which is likely the only reason to add the property declarations as well.

**Comment by weierophinney** [comment]
I re-instated the two tests that were removed when merging.

**Comment by Maks3w** [comment]
@weierophinney Those tests was not testing the trait. Those tests was testing Request constructor 

It's totally out of the scope

**Comment by weierophinney** [comment]
Then moving them to the correct test suite would have worked, but not
removing them entirely. Send a new PR if you wish to do that.

**Comment by Maks3w** [comment]
@weierophinney Please I can't fix all the test suite at once. I need to go by steps
